psc/Main.hs

```diff
@@ -52,12 +52,12 @@ data InputOptions = InputOptions
 
 compile :: PSCMakeOptions -> IO ()
 compile (PSCMakeOptions inputGlob inputForeignGlob outputDir opts usePrefix) = do
-  input <- concat <$> mapM glob inputGlob
+  input <- globWarningOnMisses (warnFileTypeNotFound "input files") inputGlob
   when (null input) $ do
     hPutStrLn stderr "psc: No input files."
     exitFailure
   moduleFiles <- readInput (InputOptions input)
-  inputForeign <- concat <$> mapM glob inputForeignGlob
+  inputForeign <- globWarningOnMisses (warnFileTypeNotFound "foreign imports") inputForeignGlob
```

Comment:
Nitpick: `foreign import` usually means a declaration. This might be confusing. I would just have both messages say "No files found using pattern:" to keep it simple. It will be obvious which type from the extension, and in any case, we're going to get rid of `--ffi` eventually anyway.